tutorials/javaFxPart5.md

```diff
@@ -0,0 +1,419 @@
+<frontmatter>
+  title: "JavaFX tutorial part 5 – Tweaking the GUI"
+  pageNav: 2
+</frontmatter>
+
+# JavaFX tutorial part 5 – Tweaking the GUI (Optional)
+
+This guide aims to help shorten the amount of time needed to learn some of the layout/CSS related stuff so that you may
+focus more on designing your own GUI.
+
+# Setting the dimensions
+
+As you may have noticed, some of the boxes does not follow the dimension of the app.
+
+To handle this, we have 3 choices:
+1) [Automatically anchor nodes](#automatically-anchor-nodes)
+2) [Set limit to window dimensions](#setting-limit-to-window-size)
+3) Both
+
+
+## Automatically Anchor Nodes
+
+We want to [anchor](https://docs.oracle.com/javase/8/javafx/api/javafx/scene/layout/AnchorPane.html) the following nodes:
+- Anchor `TextField` to the bottom while also automatically resize horizontally
+- Anchor Send Button to the bottom right of the app
+- Anchor the `ScrollPane` to resize both horizontally and vertically
+
+### Automatically Anchor TextField
+Open `src/main/resources/view/MainWindow.fxml` in SceneBuilder and click on `TextField`. On the right accordion pane you
+should be able to see the AnchorPane constraints of the `TextField`.
+
+![SceneBuilder opening MainWindow.fxml with Layout view](images/javafx/TextFieldAnchorPaneConstraints1.png)
+
+Children within `AnchorPane` are able to "anchor" their edges to the `AnchorPane` itself if a constraint is set.
+
+Since we want to automatically resize the `TextField` whenever we resize the window, we need to put an anchor
+constraint on the **left and right edges**. Click on the left and right boxes as shown below, and it will automatically
+set a value based on its current position relative to the AnchorPane. Note that the right side is set to 76 as this is
+the width of the button next to it.
+
+![Update AnchorPane constraints of TextField](images/javafx/TextFieldAnchorPaneConstraints2.png)
+
+### Automatically Anchor Send Button
+Since we want to anchor the button to the bottom right corner of the window, we
+set the anchor values for the **bottom and right** side.
+
+![Update AnchorPane constraints of Send Button](images/javafx/SendButtonAnchorPaneConstraints.png)
+
+### Automatically Anchor ScrollPane
+Since we want to resize the chat both horizontally and vertically, we set the anchor pane constraints to **all** edges.
+
+At the same time we **enable Fit To Width** so that the content within the `ScrollPane` resize as well.
+
+![Update AnchorPane constraints of ScrollPane](images/javafx/ScrollPaneAnchorPaneConstraints.png)
+
+Save `MainWindow.fxml` and you're done! Run and test your app out to see if it works when you resize the window.
```

Comment:
AFAIK, if you anchor only the `TextField`, the `ScrollPane` wont be resized. Likewise if the `ScrollPane` gets anchored, the `TextField` wont be resized unless I anchor both elements. Unless I misunderstood this.

Reply:
I see. In that case, perhaps suggest the reader to try resizing after anchoring the first element, and explain why it doesn't work fully yet? It might help the reader to better understand how the anchoring mechanism works.

Reply:
Will do!